### PR TITLE
Convert gp_sparse_vector into an extension

### PR DESCRIPTION
**Pull request by joaopapereira**
Created the needed files and also did the following:
 - Updated the installation steps
 - Cleanup Makefile (Remove commented commands and added some spacing)
 - Convert the tests that were present into the new testing framework
and added them to the REGRESS tag in the Makefile
   - Corrected some typos on the tests
   - Created 2 sets of tests one for features math and another one for
simple sparse vectors

### CODE REVIEW

**Comment by asimrp** [comment]
Couldn't spot any red flags, although I have no idea of sparse vector and the scenarios covered by the tests.  My only ask is to set the tests up such that they are run as part of `installcheck-word` target.

File: contrib/gp_sparse_vector/README_INSTALL [hunk 1]
```diff
@@ -5,5 +5,4 @@ Requirements:
 To install:
```
Comment (melanieplageman) [inline contrib/gp_sparse_vector/README_INSTALL]: Could we also update the main README to wrap text to the same standard width as the other READMEs?

**Comment by danielgustafsson** [comment]
> On 8 Aug 2018, at 22:42, Melanie <notifications@github.com> wrote:
> 
> @melanieplageman requested changes on this pull request.
> 
> Per discussion with @Eulerizeit this extension should be in gpAux/extensions instead of in contrib unless it ends up upstream

If we start moving these modules, how do we ensure they aren’t just left to bitrot (which they sort of have done in contrib/, but let’s improve on the situation)?

Looking at gpAux/extensions it consists of either vendored pieces for release engineering or modules under autoconf feature flags. We clearly don’t want autoconf flags for everything we add there, but perhaps we should gate their inclusion in “make install” in some way since they are extensions and per definition shouldn’t be automatically included? We could a single —-enable-gpdb-contrib to autoconf, or provide a target for them “make gpdb-contrib”, or something completely different. Thoughts?

This should probably be a thread on the gpdb-dev@ mailinglist..

**Comment by joaopapereira** [comment]
going to revert last commit, as per discussion on the mailing list. Eventually I will create a PR to move this extension to the agreed location

**Pull request merged**